src/pages/elements/Namespaces.js

```diff
@@ -1,17 +1,20 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
 import { Table } from 'patternfly-react';
 import GrafanaUrl from './GrafanaUrl';
 import { sortByName } from '../../components/Utils';
+import SearchBar from '../../components/SearchBar';
 
 function Namespaces({ namespaces }) {
+  const options = ['Name', 'Cluster', 'Description'];
+  const [selected, changeSelected] = useState(options[0]);
+  const [filterText, changeFilterText] = useState('');
   if (namespaces.length === 0) {
     return <p style={{ 'font-style': 'italic' }}>No namespaces.</p>;
   }
-
   const headerFormat = value => <Table.Heading>{value}</Table.Heading>;
   const cellFormat = value => <Table.Cell>{value}</Table.Cell>;
-
+  const matchedNamespaces = [];
```

Comment:
I think that instead of this approach I would've done:

```js
const lcFilter = filterText.toLowerCase();
const matchedNamespaces = processedNamespaces.filter(ns => 
        (selected === 'Name' && ns.name.toLowerCase().includes(lcFilter))  ||
        (selected === 'Cluster' && ns.cluster.name.toLowerCase().includes(lcFilter)) ||
        (selected === 'Description' && ns.description !== null && ns.description.toLowerCase().includes(lcFilter))
})
```

(you need to move it a few lines below, so `selected` and `processedNamespaces` is defined)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
I'm missing a return: `return (selected === 'Name' && ns.name.toLowerCase().includes(lcFilter)) ...`